src/GafferOSL/OSLObject.cpp

```diff
@@ -310,9 +317,10 @@ void OSLObject::hashProcessedObject( const ScenePath &path, const Gaffer::Contex
 	{
 		h.append( inPlug()->fullTransformHash( path ) );
 	}
-	if( useAttributesPlug()->getValue() )
+
+	if( useAttributes )
 	{
-		h.append( inPlug()->fullAttributesHash( path ) );
+		gafferAttributes->hash( h );
```

Comment:
We can actually do better ... I'm pretty sure we can just omit this?  We're already including `shadingEngine->hash`, which includes all the parameters that we have baked substitutions into already.

Reply:
Oh, I was just thinking about substitutions, and not thinking about how this was already needed for regular getattributes.  I have now fixed this in cab7548

In the course of implementing this, I compared it to what it was previously using, CompoundObject::hash ... looking at that was ... rather alarming?  I haven't measured yet how often Gaffer is actually using this, but I really wouldn't expect this hash to allocate a vector of iterators, and do a pile of string comparisons, just for the sake of inter-process consistency which is unnecessary for Gaffer.

Reply:
> I compared it to what it was previously using, CompoundObject::hash

It wasn't using `CompoundObject::hash()` before - it was using `fullAttributesHash()`, which is a `ValuePlug::hash()` rather than an `Object::hash()`.